src/functions-and-arrays.js

```diff
@@ -1,38 +1,131 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(a,b) {
 
+    if(a>b){
+      return a
+      console.log(a)
+    }else if (a===b ) {
+      return a
+    }else{
+      return b
+    }
 
+}
+
+console.log(maxOfTwoNumbers(5,6))
 
 // Iteration #2: Find longest word
+//                7          7           7             9       5           7        8
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+  function findLongestWord(words) {
+    var longest = 0;
+    var word = null;
 
+    if (words === undefined){
+
+      return null
+      
+    }
+
+    for (var i = 0; i < words.length ; i++) {
+        if (longest < words[i].length) {
+            longest = words[i].length;
+            word = words[i];
+        }
+    }
+    return word;
+}
+console.log(findLongestWord(words))
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  const empty = []
+  let numEmpty = 0
+
+  if (numbers === undefined){
+
+    return 0
+    
+  }
+
+  for (i=0; i< numbers.length;i++){
+
+    numEmpty += numbers[i]
+
+  }
+    return numEmpty  
+
+
+}
 
+let numEmpty = 0
+
+for (i=0; i< numbers.length;i++){
+
+  numEmpty += numbers[i]
+
+}
+console.log(numEmpty)
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum() {
+
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+  let numEmpty = 0
+  const empty = []
+  
+
+  if ( numbersAvg.length === 0) {
+    return null
+    
+  }
+
+ 
+  for (i=0; i< numbersAvg.length;i++){
+
+    numEmpty += numbersAvg[i]
+
+  }
+    return numEmpty / numbersAvg.length
```

Comment:
Buen trabajo aun que aca ya tenemos una funcion que se ocupa de hacer la suma te recomandaria hacer uso de esta para no repetir 😉 
```suggestion
    return sumNumbers(numbersAvg) / numbersAvg.length
```